tools/Packer/Program.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Compression;
+using Newtonsoft.Json;
+
+namespace Packer
+{
+    public class Program
+    {
+        private const string ConfigFile = "packer.config.json";
+
+        public static void Main(string[] args)
+        {
+            var config =
+                JsonConvert.DeserializeObject<ConfigModel>(
+                    File.ReadAllText(Path.Combine(Environment.CurrentDirectory, ConfigFile)));
+            var tempDir = GetTempDirectory();
+
+            try
+            {
+                CloneRepoAndClean(Path.Combine(tempDir, config.CloneDir), config.CloneUrl, config.PathsToRemove);
+
+                foreach (var dependency in config.GitDependencies)
+                {
+                    CloneRepoAndClean(Path.Combine(tempDir, dependency.CloneIntoDir), dependency.CloneUrl,
+                        dependency.PathsToRemove);
+                }
+
+                CreatePackage(tempDir, config);
+            }
+            catch (Exception e)
+            {
+                Console.Error.WriteLine(e.Message);
+            }
+            finally
+            {
+                Directory.Delete(tempDir, true);
+            }
+        }
+
+        private static string GetTempDirectory()
+        {
+            var tempDirPath = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+            Directory.CreateDirectory(tempDirPath);
+            return tempDirPath;
+        }
+
+
+        private static void CloneRepoAndClean(string repoPath, string cloneUrl, List<string> pathsToRemove)
+        {
+            CloneRepo(cloneUrl, repoPath);
+
+            foreach (var path in pathsToRemove)
+            {
+                var combinedPath = Path.Combine(repoPath, path);
+                Console.WriteLine($"Removing: {combinedPath}");
+                if (File.Exists(combinedPath))
+                {
+                    File.Delete(combinedPath);
+                }
+
+                if (Directory.Exists(combinedPath))
+                {
+                    //Directory.Delete(combinedPath, true);
+                    DeleteDirectoryRecursive(combinedPath);
+                }
+            }
+        }
+
+        private static void CreatePackage(string temporaryDirectory, ConfigModel config)
+        {
+            var packageName = Path.Combine(Environment.CurrentDirectory, $"{config.CloneDir}-{config.Version}.zip");
+            ZipFile.CreateFromDirectory(temporaryDirectory, packageName);
+        }
+
+        private static void CloneRepo(string cloneUrl, string repoPath)
+        {
+            var arguments = string.Join(" ", new string[] { "clone", cloneUrl, $"\"{repoPath}\"" });
+            Console.WriteLine(arguments);
+            using (var process = Process.Start("git", arguments))
```

Comment:
This doesn't redirect output, so it will fail, but you wont know why

Reply:
Fixed